tests/parser/types/numbers/test_num.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+from tests.setup_transaction_tests import chain as s, tester as t, ethereum_utils as u, check_gas, \
+    get_contract_with_gas_estimation, get_contract
+
+def test_exponents_with_nums():
+    exp_code = """
+def _num_exp(x: num, y: num) -> num:
+    return x**y
+    """
+
+    c = get_contract(exp_code)
+    assert c._num_exp(2,2) == 4
+    assert c._num_exp(2,3) == 8
+    assert c._num_exp(2,4) == 16
+    assert c._num_exp(3,2) == 9
+    assert c._num_exp(3,3) == 27
+    assert c._num_exp(72,19) == 194678249355036212415530357760196608
```

Comment:
It might be clearer to write `72**19` here. That would be asserting the result in python equals the result in viper. You could do that for the rest too.